js/ui/html_editor/modules/tableResizing.js

```diff
@@ -0,0 +1,524 @@
+                if(Math.abs(actualTableWidth - lastTableWidth) > 1) {
+                    this._tableLastWidth(frame, actualTableWidth);
+                    this._updateColumnsWidth($table, index);
+                }
+            });
+            this._updateFramesPositions();
+            this._updateFramesSeparators();
+        });
+    }
+
+    _findTables() {
+        return $(this._quillContainer).find('table');
+    }
+
+    _getWidthAttrValue($element) {
+        const attrValue = $element.attr('width');
+        return attrValue ? parseInt(attrValue) : undefined;
+    }
+
+    _tableLastWidth(frame, newValue) {
+        if(isDefined(newValue)) {
+            frame.lastWidth = newValue;
+        } else {
+            return frame?.lastWidth;
+        }
+    }
+
+    _fixTablesWidths($tables) {
+        each($tables, (index, table) => {
+            const $table = $(table);
+            const $columnElements = this._getTableDeterminantElements($table, 'horizontal');
+
+            if(!this._tableResizeFrames[index]) {
+                this._tableResizeFrames[index] = { lastWidth: undefined };
+            }
+            const frame = this._getFrameForTable[$table];
+
+            if(!frame) {
+                this._tableResizeFrames.push({ $table: $table });
+            }
+
+            if($columnElements.eq(0).attr('width')) {
+                let columnsSum = 0;
+
+                each($columnElements, (_, element) => {
+                    const $element = $(element);
+                    const columnWidth = $element.attr('width') ? this._getWidthAttrValue($element) : $element.outerWidth();
+                    columnsSum += columnWidth;
+                });
+
+                $table.css('width', 'initial');
+
+                const tableWidth = this._tableLastWidth(frame) ?? $table.outerWidth();
+
+                if(frame) {
+                    this._tableLastWidth(frame, Math.max(columnsSum, tableWidth));
+                }
+            }
+        });
+    }
+
+    _createResizeFrames($tables) {
+        this._framesForTables = new Map();
+
+        $tables.each((index, table) => {
+            const $table = $(table);
+            const $lastTable = this._tableResizeFrames[index]?.$table;
+            const $tableLastWidth = this._tableResizeFrames[index].lastWidth;
+            this._tableResizeFrames[index] = {
+                $frame: this._createTableResizeFrame(table),
+                $table: $table,
+                index: index,
+                lastWidth: ($lastTable && table === $lastTable.get(0)) ? $tableLastWidth : undefined,
+                columnsCount: this._getTableDeterminantElements($table, 'horizontal').length,
+                rowsCount: this._getTableDeterminantElements($table, 'vertical').length
+            };
+
+            this._framesForTables.set(table, this._tableResizeFrames[index]);
+        });
+
+        this._tableResizeFrames.length = $tables.length;
```

Comment:
Is it necessary? I think `length` is max index + 1.

To avoid such misunderstandings I suggest you using `push`. Is it possible here?

Reply:
it should be helpful after some tables deletion. And we cannot clear array to save lastWidth property value